### ISSUE TITLE
Support `initial_validation_split` objects

### ISSUE THREAD

**Issue by hfrick** [tidymodels/tune#697]
dev rsample gained a new interface for validation splits, `initial_validation_split()` and friends, which produce objects of class `initial_validation_split`.  The follow-up function `validation_set()` produces an `rset` like the old interface `validation_split()`. (https://github.com/tidymodels/rsample/pull/403 and https://github.com/tidymodels/rsample/pull/404)

tune should know what to do with these objects. I think this essentially means we need to adapt `last_fit()` because it takes a spilt object as input rather than an rest.

``` r
library(tidymodels) # with dev rsample

val_split <- initial_validation_split(mtcars) # new class
val_set <- validation_set(val_split) # returns the same class as the old interface

spline_rec <- recipe(mpg ~ ., data = mtcars) %>%
  step_ns(disp)
lin_mod <- linear_reg() %>%
  set_engine("lm")

# fit_resamples() and tune_*() work with rsets
spline_res <- fit_resamples(lin_mod, spline_rec, resamples = val_set,
                            control = control_resamples(save_pred = TRUE))
#> Warning in `[.tbl_df`(x, is.finite(x <- as.numeric(x))): NAs introduced by
#> coercion
try_augment_without_printing <- augment(spline_res)
#> Warning: The orginal data had 25 rows but there were 6 hold-out predictions.


spline_rec_tune <- recipe(mpg ~ ., data = mtcars) %>%
  step_ns(disp, deg_free = tune())

spline_res <- tune_grid(lin_mod, spline_rec_tune, resamples = val_set, 
                        control = control_grid(save_workflow = TRUE, save_pred = TRUE))
#> Warning in `[.tbl_df`(x, is.finite(x <- as.numeric(x))): NAs introduced by
#> coercion
#> → A | warning: prediction from a rank-deficient fit may be misleading
#> There were issues with some computations   A: x1
#> There were issues with some computations   A: x3
#> 

try_fit_best_without_printing <- fit_best(spline_res)
try_augment_without_printing <- augment(spline_res)
#> Warning: The orginal data had 25 rows but there were 6 hold-out predictions.


# this one needs adaption
spline_res <- last_fit(lin_mod, spline_rec, split = val_split)
#> Error in `new_rset()`:
#> ! Each element of `splits` must be an `rsplit` object.
#> Backtrace:
#>     ▆
#>  1. ├─tune::last_fit(lin_mod, spline_rec, split = val_split)
#>  2. └─tune:::last_fit.model_spec(lin_mod, spline_rec, split = val_split)
#>  3.   └─tune:::last_fit_workflow(wflow, split, metrics, control)
#>  4.     └─rsample::manual_rset(splits, ids = "train/test split")
#>  5.       └─rsample:::new_manual_rset(splits, ids)
#>  6.         └─rsample::new_rset(splits, ids, subclass = c("manual_rset", "rset"))
#>  7.           └─rlang::abort("Each element of `splits` must be an `rsplit` object.")
#spline_res
#augment(spline_res)
```

<sup>Created on 2023-06-30 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by simonpcouch** [comment]
I agree that you've designed this in such a way that tune and friends are well-insulated outside of those changes in `last_fit()`. :) 

Tuning functions carry around an `rset_info` attribute with information on the resampling strategy, but finetune, workflowsets, and stacks don't do anything with that info that raises a red flag for me here.

**Comment by hfrick** [comment]
Looking at how validation splits are currently handled in tune with the "old" interface of making 2 consecutive binary splits:

- `last_fit()` is given the result of `initial_split()` and thus fits on all of "not testing" (aka training and validation combined)
- `fit_best()` also combines training and validation to fit the model

We talked about that we should fit that final model only on the training data, rather than combining the training set and the validation set. The reasoning was that validation sets are usually used when the data is pretty large and then you probably don't want to combine training and validation. 

That's straightforward for `last_fit()` because it doesn't know what to do with `initial_validation_split` objects already so we can just introduce that behaviour there. `best_fit()` already works with the new interface - which means it fits the best/final model on the combination of training and validation set.

I think we should keep the behaviour of `last_fit()` and `best_fit()` consistent with each other and rather have different behaviour between "old" and "new" interface to the validation split:

- For the new interface: only use the training set
- For the old interface: combine training and validation to keep existing behaviour

This should apply to the defaults and we give `last_fit()` and `best_fit()` an argument so that users can change the default regardless of which interface they used.
